cmd/kubelet/app/server_linux.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"github.com/golang/glog"
+	"golang.org/x/exp/inotify"
+)
+
+func watchForLockfileContention(path string, done chan struct{}) error {
+	watcher, err := inotify.NewWatcher()
+	if err != nil {
+		glog.Errorf("unable to create watcher for lockfile: %v", err)
+		return err
+	}
+	if err = watcher.AddWatch(path, inotify.IN_OPEN); err != nil {
```

Comment:
Hrm, I'm not sure about `read` since we should always get the `open` event first. However, maybe we should be detecting `remove` as well?